lib/ClimaLandSimulations/src/fluxnet_simulation.jl

```diff
@@ -0,0 +1,350 @@
+    canopy_component_args = (;
+        autotrophic_respiration = autotrophic_respiration_args,
+        radiative_transfer = radiative_transfer_args,
+        photosynthesis = photosynthesis_args,
+        conductance = conductance_args,
+        hydraulics = plant_hydraulics_args,
+        energy = energy_args,
+    )
+
+    # Other info needed
+    shared_params = SharedCanopyParameters{FT, typeof(earth_param_set)}(
+        z0_m,
+        z0_b,
+        earth_param_set,
+    )
+
+    canopy_model_args = (; parameters = shared_params, domain = canopy_domain)
+
+    # Integrated plant hydraulics and soil model
+    land_input = (atmos = atmos, radiation = radiation)
+    land = SoilCanopyModel{FT}(;
+        soilco2_type = soilco2_type,
+        soilco2_args = soilco2_args,
+        land_args = land_input,
+        soil_model_type = soil_model_type,
+        soil_args = soil_args,
+        canopy_component_types = canopy_component_types,
+        canopy_component_args = canopy_component_args,
+        canopy_model_args = canopy_model_args,
+    )
+    Y, p, cds = initialize(land)
+    exp_tendency! = make_exp_tendency(land)
+
+    #Initial conditions
+    Y.soil.ϑ_l =
+        drivers.SWC.status != absent ?
+        drivers.SWC.values[1 + Int(round(t0 / DATA_DT))] : soil_ν / 2 # Get soil water content at t0
+    # Both data and simulation are reference to 2005-01-01-00 (LOCAL)
+    # or 2005-01-01-06 (UTC)
+    Y.soil.θ_i = FT(0.0)
+    T_0 =
+        drivers.TS.status != absent ?
+        drivers.TS.values[1 + Int(round(t0 / DATA_DT))] :
+        drivers.TA.values[1 + Int(round(t0 / DATA_DT))] + 40# Get soil temperature at t0
+    ρc_s =
+        volumetric_heat_capacity.(
+            Y.soil.ϑ_l,
+            Y.soil.θ_i,
+            Ref(land.soil.parameters),
+        )
+    Y.soil.ρe_int =
+        volumetric_internal_energy.(
+            Y.soil.θ_i,
+            ρc_s,
+            T_0,
+            Ref(land.soil.parameters),
+        )
+    Y.soilco2.C .= FT(0.000412) # set to atmospheric co2, mol co2 per mol air
+    ψ_stem_0 = FT(-1e5 / 9800) # pressure in the leaf divided by rho_liquid*gravitational acceleration [m] 
+    ψ_leaf_0 = FT(-2e5 / 9800)
+    ψ_comps = n_stem > 0 ? [ψ_stem_0, ψ_leaf_0] : ψ_leaf_0
+
+    S_l_ini =
+        inverse_water_retention_curve.(
+            retention_model,
+            ψ_comps,
+            plant_ν,
+            plant_S_s,
+        )
+
+    for i in 1:(n_stem + n_leaf)
+        Y.canopy.hydraulics.ϑ_l.:($i) .=
+            augmented_liquid_fraction.(plant_ν, S_l_ini[i])
+    end
+
+    Y.canopy.energy.T = drivers.TA.values[1 + Int(round(t0 / DATA_DT))] # Get atmos temperature at t0
+
+    set_initial_cache! = make_set_initial_cache(land)
+    set_initial_cache!(p, Y, t0)
+
+    # Simulation
+    sv = (;
+        t = Array{FT}(undef, length(saveat)),
+        saveval = Array{NamedTuple}(undef, length(saveat)),
+    )
+    cb = ClimaLSM.NonInterpSavingCallback(sv, saveat)

```

Comment:
There were changes made in main where we update the drivers using a callback - I think we need that here now, or the drivers wont change in time.

Reply:
done!